controllers/cloud.redhat.com/providers/iqe/impl.go

```diff
@@ -125,7 +146,93 @@ func CreateIqeJobResource(cache *providers.ObjectCache, cji *crd.ClowdJobInvocat
 	return nil
 }
 
-func createAndApplyIqeSecret(cache *providers.ObjectCache, ctx context.Context, cji *crd.ClowdJobInvocation, app *crd.ClowdApp, envName string, logger logr.Logger, client client.Client) error {
+// buildVaultEnvVars creates vault env vars for the IQE pod that will map to keys defined in the vaultSecret
+func buildVaultEnvVars(vaultSecret *core.Secret) []core.EnvVar {
+	vaultEnvVars := []core.EnvVar{
+		{Name: "DYNACONF_IQE_VAULT_LOADER_ENABLED", Value: "true"},
+		{Name: "DYNACONF_IQE_VAULT_VERIFY", Value: "true"},
+		{
+			Name: "DYNACONF_IQE_VAULT_URL",
+			ValueFrom: &core.EnvVarSource{
+				SecretKeyRef: &core.SecretKeySelector{
+					LocalObjectReference: core.LocalObjectReference{Name: vaultSecret.Name},
+					Key:                  "url",
+					Optional:             utils.BoolPtr(true),
```

Comment:
I'm passing `optional: true` on all of these to make the code simpler on the Clowder side. This way we don't have to dig into the secret and actually read it. We'll let the IQE pod parse the env var and fall back to whatever default values it would prefer to use (or abort) if one of these env vars is empty.